whylog/front/utils.py

```diff
@@ -0,0 +1,41 @@
+class FrontInput(object):
+    """
+    Object of this class represents line in the file.
+    """
+
+    def __init__(self, offset, line_content, resource_location):
+        """
+        Basic constructor of FrontInput.
+        :param offset: byte of the first character of line in the file.
+        :param line_content: the content of the line.
+        :param resource_location: obeject of ResourceLocation.
+        """
+        self.offset = offset
+        self.line_content = line_content
+        self.resource_location = resource_location
+
+
+class AbstractResourceLocation(object):
```

Comment:
This is not really an abstract class. You need to decorate it.

Also, this class does not belong in `whylog/front/utils.py`